documentation/sphinx/source/client-testing.rst

```diff
@@ -0,0 +1,204 @@
+###############
+Client Testing
+###############
+
+FoundationDB comes with its own testing framework. Tests are implemented as workloads. A workload is nothing more than a class
+that gets called by server processes running the ``tester`` role. Additionally, a ``fdbserver`` process can run a simulator that
+simulates a full fdb cluster with several machines and different configurations in one process. This simulator can run the same
+workloads you can run on a real cluster. It will also inject random failures like network partitions and disk failures.
+
+Currently, workloads can only be implemented in Java, support for other languages might come later.
+
+This tutorial explains how one can implement a workload, how one can orchestrate a workload on a cluster with multiple clients, and
+how one can run a workload within a simulator. Running in a simulator is also useful as it does not require any setup: you can simply
+run one command that will provide you with a fully functional FoundationDB cluster.
+
+Implementing a Workload
+=======================
+
+In order to implement your own workload in Java you can simply create an implementation of the abstract class ``AbstractWorkload``.
+A minimal implementation will look like this:
+
+.. code-block:: java
+
+   package my.package;
+   import com.apple.foundationdb.testing.AbstractWorkload;
+   import com.apple.foundationdb.testing.WorkloadContext;
+
+   class MinimalWorkload extends AbstractWorkload {
+       public MinimalWorkload(WorkloadContext ctx) {
+           super(ctx);
+       }
+
+       @Override
+       public void setup(Database db) {
+           log(20, "WorkloadSetup", null);
+       }
+
+       public void start(Database db) {
```

Comment:
These should have `@Override` as well, right?

Reply:
Yes, this is fixed now.